spark/tests/test_spark.py

```diff
@@ -1093,6 +1092,31 @@ def test_do_not_crash_on_single_app_failure():
         assert rest_requests_to_json.call_count == 2
 
 
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "instance,service_check",
+    [
+        (DRIVER_CONFIG, "driver"),
+        (YARN_CONFIG, "resource_manager"),
+        (MESOS_CONFIG, "mesos_master"),
+        (STANDALONE_CONFIG, "standalone_master"),
+        (STANDALONE_CONFIG_PRE_20, "standalone_master"),
+    ],
+    ids=["driver", "yarn", "mesos", "standalone", "standalone_pre_20"],
+)
+def test_no_running_apps(aggregator, dd_run_check, instance, service_check):
+    with mock.patch('requests.get', return_value=MockResponse("{}")):
+        dd_run_check(SparkCheck('spark', {}, [instance]))
+
+        # no metrics sent in this case
+        aggregator.assert_all_metrics_covered()
+        aggregator.assert_service_check(
+            'spark.{}.can_connect'.format(service_check),
+            status=SparkCheck.OK,
+            tags=['url:{}'.format(instance['spark_url'])] + CLUSTER_TAGS + instance.get('tags', []),
+        )
+
```

Comment:
Maybe we can also assert that we log a warning?

Reply:
Good idea! Added

Reply:
What do you think of using [caplog](https://docs.pytest.org/en/7.1.x/how-to/logging.html#caplog-fixture)? E.g. like in https://github.com/DataDog/integrations-core/blob/a0c720115512955c3b96e95e260446b6d407bc9f/mongo/tests/test_mongo.py#L216-L220

Reply:
That's nice, thanks! I did not know it was a thing